SingularityService/src/main/java/com/hubspot/singularity/smtp/MailTemplateHelpers.java

```diff
@@ -142,21 +158,42 @@ public MailTemplateHelpers(SandboxManager sandboxManager, SingularityConfigurati
     }
   }
 
-  private Optional<Long> getMaybeOffset(String slaveHostname, String fullPath, String filename, Optional<String> directory) {
+  // Searches through the file,
+  private Optional<Long> getMaybeOffset(final String slaveHostname, final String fullPath, final Long logLength, Optional<SingularityTask> task) {
+    long offset = 0;
+    String regex = getLogErrorRegex(task);
+    long length = logLength + regex.length(); // Get extra so that we can be sure to find the error
+    Pattern pattern = Pattern.compile(regex);
     Optional<MesosFileChunkObject> logChunkObject;
-    try {
-      logChunkObject = sandboxManager.read(slaveHostname, fullPath, Optional.<Long>absent(), Optional.<Long>absent());
-    } catch (RuntimeException e) {
-      LOG.error("Sandboxmanager failed to read {}/{} on slave {}", directory.get(), filename, slaveHostname, e);
-      return Optional.absent();
-    }
-    if (logChunkObject.isPresent()) {
-      return Optional.of(logChunkObject.get().getOffset());
-    } else {
-      LOG.error("Failed to get offset for log file {}", fullPath);
-      return Optional.absent();
+    Optional<MesosFileChunkObject> previous = Optional.absent();
+
+    while (true) { //Continue until we reach the end of the file, at which point we return from within the while loop
+      try {
+        logChunkObject = sandboxManager.read(slaveHostname, fullPath, Optional.of(offset), Optional.of(length));
+      } catch (RuntimeException e) {
+        LOG.error("Sandboxmanager failed to read {} on slave {}", fullPath, slaveHostname, e);
+        return Optional.absent();
+      }
+      if (logChunkObject.isPresent()) {
+        if (logChunkObject.get().getData().equals("")) { // Passed end of file
+          if (previous.isPresent()) { // If there was any log, get the bottom bytes of it
+            long end = previous.get().getOffset() + previous.get().getData().length();
+            return Optional.of(end - logLength);
+          }
+          return Optional.absent();
```

Comment:
returning an absent here would revert us to the previous behavior of returning the beginning portion of the file (absent offset). If we can't find anything, wouldn't we still want to see the tail like we implemented in the previous PR?